python/lib/python_port.c

```diff
@@ -85,13 +98,16 @@ PyObject* py_port_set(PyObject *self, PyObject *args) {
     }
 
     if (val) {
-        LF_PRINT_DEBUG("Setting value %p.", val);
+        LF_PRINT_DEBUG("Setting value lol %p.", val);
         Py_XDECREF(port->value);
-        Py_INCREF(val);
-        // Call the core lib API to set the port
-        _LF_SET(port, val);
 
-        Py_INCREF(val);
+        // 1. Call lf_new_token function to create a new token
+        // 2. Set destructor and assign value to the port 
+        // Py_INCREF(val) is unnecessary 
+        lf_token_t* token = lf_new_token((void*)port, val, 1);
+        lf_set_destructor(port, python_count_decrement);
```

Comment:
It's not a big deal, but setting the destructor needs to be done only once. Could it be done instead in `py_port_capsule_new`?

Reply:
Hello Edward,
Thanks for the comment. I considered moving this to the `py_port_capsule_new` function as well. However, during testing, I found that the `py_port_capsule_new` function wasn't called at all during the federated execution. Additionally, the `py_port_set` function is invoked once per iteration, and the destructor is indeed set only once.